software/muscle/main.nf

```diff
@@ -0,0 +1,59 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process MUSCLE {
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::muscle=3.8.1551" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/muscle:3.8.1551--h7d875b9_6"
+    } else {
+        container "quay.io/biocontainers/muscle:3.8.1551--h7d875b9_6"
+    }
+
+    input:
+    path fasta
```

Comment:
I think general consensus is to pass a map `val(meta)` through every module, have a look at e.g. `fastqc` how this is done.

Reply:
Thanks for pointing out - have updated